packages/tentales/src/middleware/default-hooks.ts

```diff
@@ -1,24 +1,49 @@
-import { Hook } from "tentales"
+import { Hook, Config } from "tentales"
+import { Log } from "tentales-log"
+import { path } from "ramda"
 import { bodyParserMiddleware } from "./middlewares/body-parser-middleware"
 import { renderMiddleware } from "./middlewares/render-middleware"
 import { fourOhFourMiddleware } from "./middlewares/four-oh-four-middleware"
 import { errorMiddleware } from "./middlewares/error-middleware"
 import { protectServiceRoutesMiddleware } from "./middlewares/protect-service-routes-middelware"
 
-const DEFAULT_HOOKS_PUBLIC: Hook[] = [
-  ["errorMiddleware", [errorMiddleware]],
-  ["bodyParserMiddleware", [bodyParserMiddleware]],
-  ["renderMiddleware", [renderMiddleware]],
-  ["protectServiceRoutesMiddleware", [protectServiceRoutesMiddleware]],
-  ["fourOhFourMiddleware", [fourOhFourMiddleware]],
-]
+const DEFAULT_HOOKS_PUBLIC = (log: Log, config: Config): Hook[] => {
+  const middlewares: Hook[] = [
+    ["errorMiddleware", [errorMiddleware]],
+    ["bodyParserMiddleware", [bodyParserMiddleware]],
+    ["renderMiddleware", [renderMiddleware]],
+  ]
+
+  if (path(["auth", "serverSecret"], config)) {
+    middlewares.push([
+      "protectServiceRoutesMiddleware",
+      [protectServiceRoutesMiddleware],
+    ])
+  } else {
+    log.warn(
+      "Authorization secret not set; no authorization will be carried out.",
+    )
+  }
+
+  middlewares.push(["fourOhFourMiddleware", [fourOhFourMiddleware]])
+
+  return middlewares
+}
 
 const DEFAULT_HOOKS_SERVICE: Hook[] = [
   ["errorMiddleware", [errorMiddleware]],
   ["bodyParserMiddleware", [bodyParserMiddleware]],
   ["protectServiceRoutesMiddleware", [protectServiceRoutesMiddleware]],
 ]
 
-export function getDefaultHooks({ isPublic }: { isPublic: boolean }): Hook[] {
-  return isPublic ? DEFAULT_HOOKS_PUBLIC : DEFAULT_HOOKS_SERVICE
+export function getDefaultHooks({
+  log,
+  config,
+}: {
+  log: Log
+  config: Config
+}): Hook[] {
+  return config.public
+    ? DEFAULT_HOOKS_PUBLIC(log, config)
```

Comment:
Nit/Flame war igniter: If passing more than 1 argument to a function, I've tried to pass it as an object (as we do to _this_ function), to make arguments passing explicit.

Reply:
Yeah, I remember us talking about this some time. With typescript checking the types of the arguments I figured the main benefit of sending named parameters was diminished. Sending them as normal parameters also simplify the type declaration:

```javascript
const DEFAULT_HOOKS_PUBLIC = ({log, config}: {log: Log, config: Config}) => {...}
```

vs 

```javascript
const DEFAULT_HOOKS_PUBLIC = (log: Log, config: Config) => {...}
```

Reply:
Also, note that I kept using the named parameters scheme for the exported function, and that this function is only used within this file. ^^

Reply:
TS sure does make things a bit better. My thinking is that it's still not _visible_ on screen. You still need to like hover it, click it, use a keyboard shortcut or something like it, to see the declaration. Whereas an object literal shows that straight up on your screen, no batteries needed.